src/charm.py

```diff
@@ -396,6 +399,16 @@ def _handle_status_update_config(self, event: HookEvent) -> None:
         else:
             self.unit.status = BlockedStatus("Missing postgres database relation")
 
+    def _on_install(self, event):
+        if not self._container.can_connect():
+            event.defer()
+            logger.info("Cannot connect to Kratos container. Deferring event.")
+            self.unit.status = WaitingStatus("Waiting to connect to Kratos container")
+            return
+
+        self._push_default_files()
+
+
     def _on_pebble_ready(self, event: PebbleReadyEvent) -> None:
```

Comment:
Seeing how the charm code starts to turn out, I am not sure if we need this function to be different from `_handle_status_update_config`